web-ng/src/app/components/main-page/main-page.component.ts

```diff
@@ -89,7 +89,7 @@ export class MainPageComponent implements OnInit {
     this.activeProject$.pipe(take(1)).subscribe(project =>
       this.dialog.open(LayerDialogComponent, {
         data: {
-          projectId: project.id,
+          projectId: project.isEmpty() ? ':new' : project.id,
```

Comment:
Using magic values (":new") for APIs between classes exposes implementation details across the boundaries of components (so called "leaky abstractions") and increases complexity and fragility of the system.

Since this is still a prototype, rather than design a special representation for this state, can we  move all instance of ':new' into a static const member of Project?

Reply:
Done.